src/pymovements/dataset/dataset_definition.py

```diff
@@ -39,6 +40,67 @@
 yaml.add_multi_constructor('!', type_constructor, Loader=yaml.SafeLoader)
 
 
+# sphinx automatically substitutes _Resources with the Union using the | operator.
+_Resources = Union[dict[str, list[dict[str, str]]], dict[str, tuple[dict[str, str], ...]]]
+
+
+class _HasResourcesIndexer:
```

Comment:
this helper class is necessary to allow something like:

```
definition.has_resources['gaze']
```

The `['gaze']` index is equivalent to `__getitem__('gaze')`, so the `DatasetDefinition.has_resources` property has to return an object that implements this method.

Reply:
I think having this helper function  within `dataset_definition` makes it even more intimidating to add new datasets for outside users. maybe it could go to `dataset/_utils`?

Reply:
moved to `dataset/_utils/_resources.py`